OpenRA.Mods.Common/Traits/AffectsShroud.cs

```diff
@@ -64,12 +65,20 @@ public void Tick(Actor self)
 			var projectedPos = centerPosition - new WVec(0, centerPosition.Z, centerPosition.Z);
 			var projectedLocation = self.World.Map.CellContaining(projectedPos);
 			var disabled = IsDisabled(self);
+			var traitDisabled = IsTraitDisabled;
+
+			if (traitDisabled)
+			{
+				cachedTraitDisabled = traitDisabled;
+				return;
```

Comment:
We need to `RemoveCellsFromPlayerShroud` when changing from enabled &rarr; disabled too!